File: src/dataset/javascript/etl_exercism-ThomasZumsteg/gemini/GeminiFlash_etl_v2.js
var ETL = function() {}

ETL.prototype.transform = function(oldSturct) {
	/* Converts an old data structure to a new format */
	const newStruct = {};
	for (const key in oldSturct) {
		if (oldSturct.hasOwnProperty(key)) {
			const values = oldSturct[key];
			const parsedKey = parseInt(key, 10); // Parse key once
			for (let i = 0; i < values.length; i++) {
				newStruct[values[i].toLowerCase()] = parsedKey;
			}
		}
	}
	return newStruct;
}

module.exports = ETL

File: src/dataset/cpp/etl_exercism-Akshive/gemini/GeminiFlash_etl_v2.cpp
#include "etl.h"
#include <algorithm>
#include <cctype>
#include <map>
#include <vector>

namespace etl {
    std::map<char, int> transform(const std::map<int, std::vector<char>>& old) {
        std::map<char, int> ans;
        for (const auto& [key, values] : old) {
            for (char value : values) {
                ans[std::tolower(value)] = key;
            }
        }
        return ans;
    }
}